src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -844,48 +853,143 @@ QgsExpressionContext QgsLayoutItemLegend::createExpressionContext() const
   scope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "legend_filter_out_atlas" ), legendFilterOutAtlas(), true ) );
 
   context.appendScope( scope );
-
   return context;
 }
 
+QgsExpressionContext QgsLayoutItemLegend::updateExpressionContext()
+{
+
+  QgsExpressionContext context = createExpressionContext();
+
+  mExpContext.~QgsExpressionContext();
+  const QList<QgsExpressionContextScope *> scopes = context.takeScopes();
+  for ( QgsExpressionContextScope *scopep : scopes )
+  {
+    mExpContext.appendScope( scopep );
+  }
+  return mExpContext;
+
+}
+
+
 // -------------------------------------------------------------------------
 #include "qgslayertreemodellegendnode.h"
 #include "qgsvectorlayer.h"
+#include "qgsmaplayerlegend.h"
 
 QgsLegendModel::QgsLegendModel( QgsLayerTree *rootNode, QObject *parent )
   : QgsLayerTreeModel( rootNode, parent )
 {
+  mLayoutLegendContext = nullptr;
   setFlag( QgsLayerTreeModel::AllowLegendChangeState, false );
   setFlag( QgsLayerTreeModel::AllowNodeReorder, true );
 }
 
+Qt::ItemFlags QgsLegendModel::flags( const QModelIndex &index ) const
+{
+  // make the legend nodes selectable even if they are not by default
+  if ( index2legendNode( index ) )
+    return QgsLayerTreeModel::flags( index ) | Qt::ItemIsSelectable;
+
+  return QgsLayerTreeModel::flags( index );
+}
+
 QVariant QgsLegendModel::data( const QModelIndex &index, int role ) const
```

Comment:
Modifications made to data are there to ensure that the proper label and its content are properly evaluated.